src/nrtk_explorer/app/embeddings.py

```diff
@@ -60,10 +60,11 @@ def on_feature_extraction_model_change(self, **kwargs):
 
     def on_current_dataset_change(self, **kwargs):
         self.state.num_elements_disabled = True
-        with open(self.state.current_dataset) as f:
-            dataset = json.load(f)
-            self.images = dataset["images"]
-            self.state.num_elements_max = len(self.images)
+        if self.context.dataset is None:
+            self.context.dataset = get_dataset(self.state.current_dataset, force_reload=True)
+
+        self.images = list(self.context.dataset.imgs.values())
+        self.state.num_elements_max = len(self.images)
```

Comment:
Technically it doesn't matter because len on an existing list is efficient, but if you need to know the length of the images before you iterate over them, then using `n_images` attribute will be much more efficient in an SQL context.

```python
self.state.num_elements_max = self.context.dataset.n_images
```